baybe/searchspace/continuous.py

```diff
@@ -413,22 +432,71 @@ def _sample_from_bounds(self, batch_size: int, bounds: np.ndarray) -> pd.DataFra
         return pd.DataFrame(points, columns=self.parameter_names)
 
     def _sample_from_polytope(
-        self, batch_size: int, bounds: np.ndarray
+        self,
+        batch_size: int,
+        bounds: np.ndarray,
     ) -> pd.DataFrame:
         """Draw uniform random samples from a polytope."""
+        # If the space has interpoint constraints, we need to sample from a larger
+        # searchspace that models the batch size via additional dimension. This is
+        # necessary since `get_polytope_samples` cannot handle interpoint constraints,
+        # see https://github.com/pytorch/botorch/issues/2468
+
         import torch
         from botorch.utils.sampling import get_polytope_samples
 
+        # The number of parameters is needed at some places for adjusting indices
+        num_of_params = len(self.parameters)
+
+        eq_constraints, ineq_constraints = [], []
+
+        # We start with the general constraints before going to interpoint constraints
```

Comment:
this comment implies there is a loop over the former, then over the latter, but in reality its one loop combined so its a bit confusing